File: back/DevC/src/main/java/com/back/devc/domain/interaction/notification/entity/Notification.java
package com.back.devc.domain.interaction.notification.entity;

import com.back.devc.global.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "notifications")
public class Notification extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "notification_id")
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "actor_user_id", nullable = false)
    private Long actorUserId;

    @Column(name = "post_id")
    private Long postId;

    @Column(name = "comment_id")
    private Long commentId;

    @Column(name = "type", nullable = false, length = 30)
    private String type;

    @Column(name = "message", nullable = false, length = 255)
    private String message;

    @Column(name = "is_read", nullable = false)
    private boolean isRead;

    protected Notification() {
    }

    public Notification(Long userId, Long actorUserId, Long postId, Long commentId, String type, String message) {
        this.userId = userId;
        this.actorUserId = actorUserId;
        this.postId = postId;
        this.commentId = commentId;
        this.type = type;
        this.message = message;
        this.isRead = false;
    }

    public static Notification create(Long userId, Long actorUserId, Long postId, Long commentId, String type, String message) {
        return new Notification(userId, actorUserId, postId, commentId, type, message);
    }

    public void markAsRead() {
        this.isRead = true;
    }

    public Long getId() {
        return id;
    }

    public Long getUserId() {
        return userId;
    }

    public Long getActorUserId() {
        return actorUserId;
    }

    public Long getPostId() {
        return postId;
    }

    public Long getCommentId() {
        return commentId;
    }

    public String getType() {
        return type;
    }

    public String getMessage() {
        return message;
    }

    public boolean isRead() {
        return isRead;
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/interaction/notification/entity/Notification.kt
package com.back.devc.domain.interaction.notification.entity

import com.back.devc.global.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "notifications")
class Notification protected constructor() : BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "notification_id")
    var id: Long? = null
        protected set

    @Column(name = "user_id", nullable = false)
    var userId: Long? = null
        protected set

    @Column(name = "actor_user_id", nullable = false)
    var actorUserId: Long? = null
        protected set

    @Column(name = "post_id")
    var postId: Long? = null
        protected set

    @Column(name = "comment_id")
    var commentId: Long? = null
        protected set

    @Column(name = "type", nullable = false, length = 30)
    var type: String? = null
        protected set

    @Column(name = "message", nullable = false, length = 255)
    var message: String? = null
        protected set

    @Column(name = "is_read", nullable = false)
    var isRead: Boolean = false
        protected set

    constructor(
        userId: Long?,
        actorUserId: Long?,
        postId: Long?,
        commentId: Long?,
        type: String?,
        message: String?,
    ) : this() {
        this.userId = userId
        this.actorUserId = actorUserId
        this.postId = postId
        this.commentId = commentId
        this.type = type
        this.message = message
        this.isRead = false
    }

    fun markAsRead() {
        this.isRead = true
    }

    companion object {
        @JvmStatic
        fun create(
            userId: Long?,
            actorUserId: Long?,
            postId: Long?,
            commentId: Long?,
            type: String?,
            message: String?,
        ): Notification {
            return Notification(
                userId,
                actorUserId,
                postId,
                commentId,
                type,
                message,
            )
        }
    }
}